corehq/apps/reports/views.py

```diff
@@ -1166,26 +1184,53 @@ def _render_report_configs(request, configs, domain, owner_id, couch_user, email
             "enddate": date_range.get("enddate") if date_range else "",
         })
 
-    return render(request, "reports/report_email.html", {
+    return render(request, "reports/report_email_content.html", {
         "reports": report_outputs,
         "domain": domain,
         "couch_user": owner_id,
         "DNS_name": get_url_base(),
         "owner_name": couch_user.full_name or couch_user.get_email(),
         "email": email,
         "notes": notes,
-        "unsub_link": get_url_base() + reverse('notification_unsubscribe', kwargs={
-            'scheduled_report_id': report_id, 'domain': domain}) if report_id else None,
         "report_type": _("once off report") if once else _("scheduled report"),
-    }), excel_attachments
+    }).content, excel_attachments
+
+
+def render_full_report_notification(request, content, email=None, report=None):
+    """
+    Renders full notification body with provided main content.
+    """
+    from dimagi.utils.web import get_url_base
+    from django.http import HttpRequest
+
+    if request is None:
+        if report is None:
+            raise ValueError("Can't render notification without providing request or report")
+        request = HttpRequest()
+        request.couch_user = report.couch_user
+        request.user = request.couch_user.get_django_user()
+        request.domain = report.domain
+        request.couch_user.current_domain = report.domain
```

Comment:
Doesn't seem like you tested this at all. This fails when called from celery because `report` has not attribute `couch_user`.

I don't think the request here needs the `couch_user`, `user` or domain `domain` fields.